### DISCUSSION TITLE
Question about 3D Graph for PCQM4Mv2

### DISCUSSION THREAD

**Discussion by yihhhh** [snap-stanford/ogb#276] [PCQM4M-LSC]
Hi @weihua916, our team is trying to use the 3D graph information in our model, however, when I try to find the corresponding 3D graph for the original 2D graph though index, I find that the 2D graph node number processed thought smlie2graph is not consistent with the atom number in 3D graph. For instance, when index=0, the molecule is `O=C1[N]c2ccncc2[CH][C@@H]1c1ccc(cc1)C`, which contains 18 nodes, while 0.xyz provides 3D coordinates of 30 atoms.

**Comment by weihua916** [comment]
You need to add hydrogen.

```
>>> from rdkit import Chem
>>> smiles = 'O=C1[N]c2ccncc2[CH][C@@H]1c1ccc(cc1)C'
>>> m = Chem.MolFromSmiles(smiles)
>>> m2 = Chem.AddHs(m)
>>> num_atoms_smiles = m2.GetNumAtoms()
>>> num_atoms_smiles
30
```

**Reply by xfcui** [reply]
Unfortunately, that does not solve the problem completely. Although the number of atoms are matched, the atom-coordinate matching is still missing. For example, given the following line from 0.xyz, which C atom of the SMILES string "O=C1[N]c2ccncc2[CH][C@@H]1c1ccc(cc1)C" is it mapping to?

C   4.9918889297 -5.2514177104 4.0125716343

**Reply by weihua916** [reply]
The atom-coordinate matching is inherently difficult. See https://github.com/snap-stanford/ogb/issues/297

**Reply by weihua916** [reply]
Hi! We have just uploaded the SDF file, from which you can create 2D graphs with XYZ coordinate.
Please see [here](https://ogb.stanford.edu/docs/lsc/pcqm4mv2/) for the tutorial on how to use the SDF file.
